graphql_compiler/compiler/metadata.py

```diff
@@ -132,6 +151,18 @@ def get_location_info(self, location):
                                  u'{}'.format(location))
         return location_info
 
+    def record_explain_info(self, location, explain_info):
+        """Record estimation information about the location."""
+        if isinstance(location, FoldScopeLocation):
+            # NOTE(gurer): ignore filters inside the fold for now
+            return
+        record_location = location.at_vertex()
+        self._explain_infos[record_location].append(explain_info)
+
+    def get_explain_infos(self, location):
+        """Get estimation information of the location."""
+        return self._explain_infos[location]
```

Comment:
A custom error message might be nice here instead of failing on a `KeyError`.

Reply:
Filter info is optional for most locations. It is easier to get [] from callers perspective than to check for exception or none.